pkg/controller/command/vdri/command.go

```diff
@@ -30,41 +34,72 @@ const (
 
 	// CreatePublicDIDError is for failures while creating public DIDs
 	CreatePublicDIDError
+
+	// SaveDIDErrorCode for save did error
+	SaveDIDErrorCode
+
+	// GetDIDErrorCode for get did error
+	GetDIDErrorCode
+
+	// GetDIDByNameErrorCode for get did  by name error
+	GetDIDByNameErrorCode
 )
 
 const (
 	// command name
 	commandName = "vdri"
 
+	// command methods
+	createPublicDIDCommandMethod = "CreatePublicDID"
+	saveDIDCommandMethod         = "SaveDID"
+	getDIDsCommandMethod         = "GetDIDs"
+	getDIDCommandMethod          = "GetDID"
+	getDIDByNameCommandMethod    = "GetDIDByName"
+
 	// error messages
 	errDIDMethodMandatory = "invalid method name"
+	errEmptyDIDName       = "did name is mandatory"
+	errEmptyDIDID         = "did id is mandatory"
 
-	// command methods
-	createPublicDIDCommandMethod = "CreatePublicDID"
+	// log constants
+	didID   = "didID"
+	didName = "didName"
 )
 
 // provider contains dependencies for the vdri controller command operations
 // and is typically created by using aries.Context()
 type provider interface {
 	VDRIRegistry() vdriapi.Registry
+	StorageProvider() storage.Provider
 }
 
 // Command contains command operations provided by vdri controller
 type Command struct {
-	ctx provider
+	ctx      provider
+	didStore *didstore.Store
 }
 
 // New returns new vdri controller command instance
-func New(ctx provider) *Command {
-	return &Command{
-		ctx: ctx,
+func New(ctx provider) (*Command, error) {
+	didStore, err := didstore.New(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("new did store : %w", err)
 	}
+
+	return &Command{
+		ctx:      ctx,
+		didStore: didStore,
+	}, nil
 }
 
 // GetHandlers returns list of all commands supported by this controller command
 func (o *Command) GetHandlers() []command.Handler {
 	return []command.Handler{
 		cmdutil.NewCommandHandler(commandName, createPublicDIDCommandMethod, o.CreatePublicDID),
+		cmdutil.NewCommandHandler(commandName, saveDIDCommandMethod, o.SaveDID),
+		cmdutil.NewCommandHandler(commandName, getDIDCommandMethod, o.GetDID),
```

Comment:
GetDID is returning DID 
& 
GETDID by name is returning record?

Then names of commands are confusing.

Reply:
GetDIDByName basically fetchs the did ID by the name , then you return back the ID based on the name as input. Get DID  has the id so you fetch the record based on the ID

Reply:
I have removed this function since its not required